src/main/scala/org/apache/spark/sql/pulsar/PulsarProvider.scala

```diff
@@ -368,6 +378,23 @@ private[pulsar] object PulsarProvider extends Logging {
     }
   }
 
+  private def getAdminParams(parameters: Map[String, String]): Map[String, String] = {
```

Comment:
This method is copied multiple times. Could you just merge these three `getProducerParams`, `getReaderParams`, `getAdminParams` method into one common method.

Reply:
That is a good idea indeed, fixed this.

Reply:
That's a good idea indeed, fixed this.